### PR TITLE
If feature groups is more than 10, short it to 10 with "Other" for the rest

### PR DESCRIPTION
**Pull request by RubyZ10**
Signed-off-by: Ruby Zhu <zhenzhu@microsoft.com>

## Description

This PR is to fix bug [Bug 1874148](https://msdata.visualstudio.com/Vienna/_workitems/edit/1874148): Model Overview: feature group is too long for first_name and last_name where it contains a lot of categories

When the feature groups is more than 10, keep the top 10 and have the rest as other under the rows

Before
image

After
image

## Checklist

- [x] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/FeaturesUtils.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { localization } from "@responsible-ai/localization";
```
Comment (imatiach-msft) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/FeaturesUtils.ts:1]: looks like there are some linting errors, you can run yarn lintfix to autofix them or see those which can't be automatically fixed

**Review by imatiach-msft — APPROVED** [review]

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/FeaturesUtils.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  // if feature groups is more than 10, keep the top 10 groups, for the rest show as Other
+  items.forEach((item) => {
+    if (item.groups.length > 10) {
+      const groups = item.groups.slice(0, 10);
+      groups.push(localization.ModelAssessment.ModelOverview.other);
+      item.groups = groups;
+    }
+  });
+  return items;
+}
```
Comment (gaugup) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/FeaturesUtils.ts:18]: Could you add a simple unit test for this change?

**Review by tongyu-microsoft — APPROVED** [review]

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/FeaturesUtils.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+// Licensed under the MIT License.
+
+import { localization } from "@responsible-ai/localization";
+
+import { IFeatureConfigurationRow } from "./FeatureConfigurationFlyout";
+
+export function shortFeatureGroups(items: IFeatureConfigurationRow[]) {
+  // if feature groups is more than 10, keep the top 10 groups, for the rest show as Other
+  items.forEach((item) => {
+    if (item.groups.length > 10) {
```
Comment (vinuthakaranth) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/FeaturesUtils.ts:11]: nit: should this be a constant?

**Review by vinuthakaranth — APPROVED** [review]

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/FeaturesUtils.ts (hunk 2 above)
Comment (RubyZ10) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/FeaturesUtils.ts:20]: added unit tests, just to double confirm you do mean unit tests not e2e test, right? (as I don't see many unit tests just to confirm)

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/FeaturesUtils.ts (hunk 3 above)
Comment (RubyZ10) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/FeaturesUtils.ts]: added thanks

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/FeaturesUtils.ts (hunk 2 above)
Comment (gaugup) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/FeaturesUtils.ts:20]: yes, I meant unit tests. Thanks for adding.

**Review by gaugup — APPROVED** [review]

**Pull request merged**